nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/blob/AbstractBlobStore.java

```diff
@@ -101,6 +115,76 @@ public String writeBlob(BlobContext blobContext) throws IOException {
         return writeBlob(blobWriteContext);
     }
 
+    @Override
+    public String writeBlob(BlobWriteContext blobWriteContext) throws IOException {
+        String returnedKey = writeBlobUsingOptimizedCopy(blobWriteContext);
+        if (returnedKey != null) {
+            return returnedKey;
+        }
+        return writeBlobGeneric(blobWriteContext);
+    }
+
+    /**
+     * Writes the blob without using any store-to-store optimization.
+     *
+     * @since 11.5
+     */
+    protected abstract String writeBlobGeneric(BlobWriteContext blobWriteContext) throws IOException;
```

Comment:
Should we put a default implementation for the backport?

Reply:
The new default implem `AbstractBlobStore.writeBlob(BlobWriteContext)` is the only one that calls `writeBlobGeneric`. And an old customer `AbstractBlobStore` subclass had to have an existing `writeBlob(BlobWriteContext)` (because it's in the interface), so any existing customer implementation will continue to work and never call `writeBlobGeneric`. So I think we're ok.

Reply:
Hm OTOH it's `abstract` so recompilation would fail... Ok I'll add a default implementation throwing something.